app/services/punches_service.rb

```diff
@@ -0,0 +1,20 @@
+class PunchesService
+  def self.create(punches, user)
```

Comment:
This method is kinda anemic at the same time you have another method doing two different things, what do you think about keeping specialized methods? Suggestion:

```ruby
class CreatePunchService
  class << self
    def call(collection_of_punches, user)
      days_to_delete = collection_of_punches.map { |punch| punch['from'] }

      ActiveRecord::Base.transaction do
        delete_punches_for_the_same_day(user, days_to_delete)
        create_new_punch(user, collection_of_punches)
      end
    end

    private

    def delete_punches_for_the_same_day(user, days_to_delete)
      user.punches.by_days(days_to_delete).delete_all
    end

    def create_new_punch(user, collection_of_punches)
      users.punches.where(
        company: user.company
      ).create(collection_of_punches)
    end
  end
end
```

Reply:
Cool! This blow up my mind kkkkk, I didnt know how to stay with the transactions block with 2 methods

Reply:
These two methods are going to be part of the transaction considering you're invoking them inside the transaction block, anyone that raises an error will trigger the rollback action.